lib/DojoAMDMainTemplate.runtime.js

```diff
@@ -115,10 +110,14 @@ module.exports = function() {
 		}
 		if (type === '[object Array]') {
 			var modules = [], callback = a2;
-			a1.forEach(function(mid) {
-				modules.push(findModule(mid, referenceModule));
-			});
-			callback.apply(this, modules);
+			try {
```

Comment:
In order to more closely mimic the Dojo behavior, I think that the try/catch should be around the `findModule()` call.  With Dojo, if a module can't be loaded, all the other modules in the dependency array are still loaded and initialized.  So we should not abandon looping though the dependency array at the first failure.  Instead, we should remember that the failure occurred, and continue calling `findModule()` for the rest of the modules in the array.  Once we've attempted to load all the modules, then we call the callback if there were no errors.

Reply:
As an example of where this would make a difference, consider the following code:

```javascript
require(["missing", "existing"], function() {
   console.log("should not get here");
});
var existing = require("existing");  // should not throw
```

The way you currently have it, the call to synchronous require would throw because the module would not have been initialized.

Reply:
Yup, I can see why its needed.